vignettes/admiral.Rmd

```diff
@@ -83,22 +89,28 @@ input and return a vector. These function can be used in expressions like
 `convert_dtc_to_dt()` in the derivation of `FINLABDT` in the
 example below:
 
-```r
-# derive treatment variables
+```{r, eval=TRUE}
+# keep final lab visit records
+ds_final_lab_visit <- ds %>% 
+  filter(DSDECOD == "FINAL LAB VISIT")
+
+# derive treatment variables  
 adsl <- dm %>%
   mutate(TRT01P = ARMCD, TRT01A = ACTARMCD) %>%
   derive_var_trtsdtm(dataset_ex = ex) %>%
-  mutate(TRTSDT = date(TRTSDTM)) %>%
+  derive_vars_dtm_to_dt(vars(TRTSDTM)) %>%
   derive_var_trtedtm(dataset_ex = ex) %>%
-  mutate(TRTEDT = date(TRTEDTM)) %>%
+  derive_vars_dtm_to_dt(vars(TRTEDTM)) %>%
   derive_var_trtdurd() %>%
   # derive final lab visit date
-  left_join(ds %>% 
-              filter(DSDECOD == "FINAL LAB VISIT") %>%
+  left_join(ds_final_lab_visit %>% 
               transmute(USUBJID, FINLABDT = convert_dtc_to_dt(DSSTDTC)),
```

Comment:
Could you move the `transmute()` call up to where you define `ds`?

Reply:
Done.